jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenLayerConfigurations.java

```diff
@@ -91,6 +111,73 @@ static JavaLayerConfigurations getForProject(
     return layerBuilder.build();
   }
 
+  /**
+   * Resolves the source files configuration for a War {@link MavenProject}.
+   *
+   * @param project the {@link MavenProject}
+   * @param extraDirectory path to the directory for the extra files layer
+   * @param appRoot root directory in the image where the app will be placed
+   * @return a {@link JavaLayerConfigurations} for the project
+   * @throws IOException if collecting the project files fails
+   */
+  private static JavaLayerConfigurations getForWarProject(
+      MavenProject project, Path extraDirectory, AbsoluteUnixPath appRoot) throws IOException {
+
+    Path explodedWarPath =
```

Comment:
Let's add a TODO to explode the WAR file rather than using this directory. The contents of the final WAR may be different from this directory. (It's possible to include or exclude files when packaging a WAR.)

Reply:
Done

Reply:
Let's also add the comment at end:

"Also the exploded WAR directory is configurable with <webappDirectory> and may not be at build.getFinalName().

Reply:
OK

Reply:
Yes of course.